frontend/src/scenes/insights/Insight.tsx

```diff
@@ -72,6 +73,9 @@ export function Insight({ insightId }: { insightId: InsightShortId | 'new' }): J
 
     useEffect(() => {
         reportInsightViewedForRecentInsights()
+        return () => {
+            abortAnyRunningQuery()
+        }
     }, [insightId])
```

Comment:
What effect does this useEffect depending on insightId have?

Should we make it agnostic?